EIPS/eip-2315.md

```diff
@@ -13,55 +13,67 @@ requires: 3540, 3670, 4200
 
 ## Abstract
 
-This proposal provides a _complete_, _safe_ and _efficient_ control-flow facility.
+This proposal provides a _complete_, _static_, _safe_ and _efficient_ control-flow facility.
 
 It deprecates `JUMP` and `JUMPI`.
 
-It introduces two opcodes to support simple subroutines:
-* `RJUMPSUB destination`— call a subroutine
-* `RETURNSUB` — return from call
-
 It depends on the two opcodes proposed by [EIP-4200](./eip-4200.md):
-* `RJUMP destination` — relative jump
-* `RJUMPI destination` — conditional relative jump
+* `RJUMP relative_offset` — relative jump to `PC + relative_offset`
+* `RJUMPI relative_offset` — conditional relative jump
+
+It introduces two opcodes to support calling and returning from subroutines:
+* `RJUMPSUB relative_offset` -- relative jump to subroutine
+* `RETURNSUB` -- return to `PC` after most recent `RJUMPSUB`.
 
-In concert with [EIP-3540](./eip-4200.md) and [EIP-3670](./eip-3670.md) it ensures that valid code will not execute invalid instructions, jump to invalid locations, or underflow stack. It will maintain consistent numbers of inputs and outputs for subroutines, and ensure that stack height is bounded in the absence of recursion.
+In concert with [EIP-3540](./eip-4200.md) and [EIP-3670](./eip-3670.md) it ensures, at initialization time, that valid code will not execute invalid instructions or jump to invalid locations, will not underflow stack, will maintain consistent numbers of inputs and outputs for subroutines, and will have bounded stack height in the absence of recursion.
```

Comment:
```suggestion
In concert with [EIP-4200](./eip-4200.md) and [EIP-3670](./eip-3670.md) it ensures, at initialization time, that valid code will not execute invalid instructions or jump to invalid locations, will not underflow stack, will maintain consistent numbers of inputs and outputs for subroutines, and will have bounded stack height in the absence of recursion.
```

Reply:
EIP-3540 -> EIP-4200 to be match with filename.

Reply:
I think I gout the filename wrong.